packages/web/app/src/pages/target-laboratory.tsx

```diff
@@ -328,8 +328,26 @@ function LaboratoryPageContent(props: {
         });
         try {
           const result = await preflightScript.execute();
-          if (result && headers) {
-            headers = substituteVariablesInHeader(headers, result);
+          // todo: Why check `result` truthiness here if it can never by falsy?
+          if (result) {
+            // We merge the result headers into the fetcher headers before performing header variable substitution
+            // so that users can have a predictable experience working with headers regardless of the place those
+            // headers come from.
```

Comment:
I think any behaviour is fine as long as it is explained within the documentation.

The reason people asked for this feature is to avoid having to substitute headers and can set these directly.
So to me it might even make sense to apply these headers after substitution.

Reply:
Despite the comment, I do not have a strong opinion on the product behaviour here.

Please let me know your final position and I will make it so.

> I think any behaviour is fine as long as it is explained within the documentation.

Agreed, I have made a note about documenting this in the PR task list.

Reply:
https://github.com/graphql/graphiql/blob/main/packages/graphiql-toolkit/src/create-fetcher/types.ts#L107

Reply:
I took your "might" as a preference and made the change in that direction.